// vite-plugin-svgr/dist/index.js
import { createFilter } from "@rollup/pluginutils";
import fs from "node:fs";
//#region src/index.ts
function vitePluginSvgr({ svgrOptions, esbuildOptions, oxcOptions, include = "**/*.svg?react", exclude } = {}) {
	const filter = createFilter(include, exclude);
	const postfixRE = /[?#].*$/s;
	return {
		name: "vite-plugin-svgr",
		enforce: "pre",
		async load(id) {
			if (!filter(id)) return;
			const filePath = id.replace(postfixRE, "");
			const svgCode = await fs.promises.readFile(filePath, "utf8");
			const { transform: svgrTransform } = await import("@svgr/core");
			const { default: jsx } = await import("@svgr/plugin-jsx");
			const componentCode = await svgrTransform(svgCode, svgrOptions, {
				filePath,
				caller: { defaultPlugins: [jsx] }
			});
			if ((this?.meta)?.rolldownVersion != null) {
				/* c8 ignore next */
				const { transformWithOxc } = await import("vite");
				return {
					code: (await transformWithOxc(componentCode, id, {
						lang: "jsx",
						...oxcOptions
					})).code,
					map: null
				};
			}
			/* c8 ignore next */
			const { transformWithEsbuild } = await import("vite");
			return {
				code: (await transformWithEsbuild(componentCode, id, {
					loader: "jsx",
					...esbuildOptions
				})).code,
				map: null
			};
		}
	};
}
//#endregion
export { vitePluginSvgr as default };
