// vite-plugin-svgr/dist/index.cjs
//#region \0rolldown/runtime.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __copyProps = (to, from, except, desc) => {
	if (from && typeof from === "object" || typeof from === "function") for (var keys = __getOwnPropNames(from), i = 0, n = keys.length, key; i < n; i++) {
		key = keys[i];
		if (!__hasOwnProp.call(to, key) && key !== except) __defProp(to, key, {
			get: ((k) => from[k]).bind(null, key),
			enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
		});
	}
	return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", {
	value: mod,
	enumerable: true
}) : target, mod));
//#endregion
let _rollup_pluginutils = require("@rollup/pluginutils");
let node_fs = require("node:fs");
node_fs = __toESM(node_fs);
//#region src/index.ts
function vitePluginSvgr({ svgrOptions, esbuildOptions, oxcOptions, include = "**/*.svg?react", exclude } = {}) {
	const filter = (0, _rollup_pluginutils.createFilter)(include, exclude);
	const postfixRE = /[?#].*$/s;
	return {
		name: "vite-plugin-svgr",
		enforce: "pre",
		async load(id) {
			if (!filter(id)) return;
			const filePath = id.replace(postfixRE, "");
			const svgCode = await node_fs.default.promises.readFile(filePath, "utf8");
			const { transform: svgrTransform } = await import("@svgr/core");
			const { default: jsx } = await import("@svgr/plugin-jsx");
			const componentCode = await svgrTransform(svgCode, svgrOptions, {
				filePath,
				caller: { defaultPlugins: [jsx] }
			});
			if ((this?.meta)?.rolldownVersion != null) {
				/* c8 ignore next */
				const { transformWithOxc } = await import("vite");
				return {
					code: (await transformWithOxc(componentCode, id, {
						lang: "jsx",
						...oxcOptions
					})).code,
					map: null
				};
			}
			/* c8 ignore next */
			const { transformWithEsbuild } = await import("vite");
			return {
				code: (await transformWithEsbuild(componentCode, id, {
					loader: "jsx",
					...esbuildOptions
				})).code,
				map: null
			};
		}
	};
}
//#endregion
module.exports = vitePluginSvgr;
